extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -323,16 +321,19 @@ export abstract class AzureAuth implements vscode.Disposable {
 			Logger.error(`Error: Could not fetch the azure resource ${azureResource} `);
 			return null;
 		}
+		// Resource endpoint must end with '/' to form a valid scope for MSAL token request.
+		const endpoint = resource.endpoint.endsWith('/') ? resource.endpoint : resource.endpoint + '/';
+
 		let account: AccountInfo | null = await this.getAccountFromMsalCache(accountId);
 		if (!account) {
 			Logger.error('Error: Could not fetch account when acquiring token');
 			return null;
 		}
 		let newScope;
 		if (resource.azureResourceId === azdata.AzureResource.ResourceManagement) {
-			newScope = [`${resource?.endpoint}user_impersonation`];
+			newScope = [`${endpoint}user_impersonation`];
 		} else {
-			newScope = [`${resource?.endpoint}.default`];
+			newScope = [`${endpoint}.default`];
```

Comment:
So was looking at this thing and noticed that it says in some situations it needs to have a double slash - https://learn.microsoft.com/en-us/azure/active-directory/develop/scenario-daemon-acquire-token?tabs=dotnet#azure-ad-v10-resources

Is that not something we have to worry about for our stuff?

Reply:
I'm not sure what would be a "resource that accepts a version 1.0 access token" since we fetch 2.0 access tokens with MSAL.

Reply:
Understanding from [this comment](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/747#issuecomment-451047774), leads me to think this should not apply to us, since we fetch 2.0 access tokens. 
Also referenced: https://nicolgit.github.io/AzureAD-Endopoint-V1-vs-V2-comparison/

Reply:
Ah ok, so does MSAL always get v2.0 tokens? Or is that a configuration we have on our side to guarantee that?

Reply:
Yes, MSAL fetches from v2.0 endpoints only. Quoting from [Documentation](https://learn.microsoft.com/en-us/azure/active-directory/develop/msal-overview#migrate-apps-that-use-adal-to-msal):

> Active Directory Authentication Library (ADAL) integrates with the Azure AD for developers (v1.0) endpoint, where MSAL integrates with the Microsoft identity platform. The v1.0 endpoint supports work accounts, but not personal accounts. The v2.0 endpoint is the unification of Microsoft personal accounts and work accounts into a single authentication system. Additionally, with MSAL you can also get authentications for Azure AD B2C.